js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx

```diff
@@ -162,25 +250,126 @@ export function buildAssetNodeStatusContent({
 
   const materializingRunId = inProgressRunIds[0] || unstartedRunIds[0];
   const late = isAssetLate(liveData);
+  const currentMinutesLate = liveData.freshnessInfo?.currentMinutesLate || 0;
 
   if (definition.isSource) {
     if (materializingRunId) {
+      return {case: StatusCase.SOURCE_OBSERVING, materializingRunId, liveData};
+    }
+    if (liveData?.lastObservation) {
+      return {
+        case: StatusCase.SOURCE_OBSERVED,
+        liveData,
+        lastObservation: liveData.lastObservation,
```

Comment:
I'm not sure slicing this down to a subset of the properties is that helpful unless we do some meaningful summarization or transformation. Maybe just having the status cases and then rendering them using the available data would be more straightforward? Don't feel too strongly about it though!

Reply:
I can only return the ones that have been calculated so far. For example for loading, liveData would be null, so if I were to share the type between all of the cases it would have to be nullable. Basically this is a way to reuse any type refinements available for the case